flink-table/flink-table-common/src/main/java/org/apache/flink/table/annotation/FunctionHint.java

```diff
@@ -157,25 +165,14 @@
     boolean isVarArgs() default false;
 
     /**
-     * Explicitly lists the argument names that a function takes as input.
+     * Explicitly lists the arguments that a function takes as input. Including their names, data
+     * types, kinds, and whether they are optional.
      *
-     * <p>By default, if {@link #input()} is defined, explicit argument names are undefined and this
-     * parameter can be used to provide argument names. If {@link #input()} is not defined, the
-     * reflection-based extraction is used, thus, this parameter is ignored.
+     * <p>It is recommended to use this parameter instead of {@link #input()}. Using both {@link
+     * #input()} and this parameter is not allowed. Specifying the list of arguments manually
+     * disables the entire reflection-based extraction around arguments.
      */
-    String[] argumentNames() default {""};
-
-    /**
-     * Explicitly lists the argument that a function takes as input, including their names, types,
-     * and whether they are optional.
-     *
-     * <p>By default, it is recommended to use this parameter instead of {@link #input()}. If the
-     * type of argumentHint is not defined, it will be considered an invalid argument and an
-     * exception will be thrown. Additionally, both this parameter and {@link #input()} cannot be
-     * defined at the same time. If neither argument nor {@link #input()} are defined,
-     * reflection-based extraction will be used.
-     */
-    ArgumentHint[] argument() default {};
+    ArgumentHint[] arguments() default {};
```

Comment:
I think it also requires changing the doc for both en and cn locales e.g. 
https://github.com/apache/flink/blob/0c338ae6d21e6d3dc79764e7801cbb7adfe6a44f/docs/content.zh/docs/dev/table/procedures.md?plain=1#L401
https://github.com/apache/flink/blob/0c338ae6d21e6d3dc79764e7801cbb7adfe6a44f/docs/content.zh/docs/dev/table/functions/udfs.md?plain=1#L664